packages/cli/src/endpoints/credentials.endpoints.ts

```diff
@@ -0,0 +1,385 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable import/no-cycle */
+import { getConnection, In } from 'typeorm';
+import { UserSettings, Credentials } from 'n8n-core';
+import { INodeCredentialTestResult } from 'n8n-workflow';
+
+import {
+	CredentialsHelper,
+	Db,
+	GenericHelpers,
+	ICredentialsDb,
+	ICredentialsResponse,
+	ResponseHelper,
+	whereClause,
+} from '..';
+import { RESPONSE_ERROR_MESSAGES } from '../constants';
+import { CredentialsEntity } from '../databases/entities/CredentialsEntity';
+import { SharedCredentials } from '../databases/entities/SharedCredentials';
+import { validateEntity } from '../GenericHelpers';
+import type { CredentialRequest } from '../requests';
+import type { N8nApp } from '../UserManagement/Interfaces';
+
+export function credentialsEndpoints(this: N8nApp): void {
+	/**
+	 * Generate a unique credential name.
+	 */
+	this.app.get(
+		`/${this.restEndpoint}/credentials/new`,
+		ResponseHelper.send(async (req: CredentialRequest.NewName): Promise<{ name: string }> => {
+			const { name: newName } = req.query;
+
+			return GenericHelpers.generateUniqueName(
+				newName ?? this.defaultCredentialsName,
+				'credentials',
+			);
+		}),
+	);
+
+	/**
+	 * Test a credential.
+	 */
+	this.app.post(
+		`/${this.restEndpoint}/credentials-test`,
+		ResponseHelper.send(async (req: CredentialRequest.Test): Promise<INodeCredentialTestResult> => {
+			const { credentials, nodeToTestWith } = req.body;
+
+			const encryptionKey = await UserSettings.getEncryptionKey();
+
+			if (!encryptionKey) {
+				return {
+					status: 'Error',
+					message: RESPONSE_ERROR_MESSAGES.NO_ENCRYPTION_KEY,
+				};
+			}
+
+			const helper = new CredentialsHelper(encryptionKey);
+
+			return helper.testCredentials(credentials.type, credentials, nodeToTestWith);
+		}),
+	);
+
+	/**
+	 * Create a credential.
+	 */
+	this.app.post(
+		`/${this.restEndpoint}/credentials`,
+		ResponseHelper.send(async (req: CredentialRequest.Create) => {
+			delete req.body.id; // delete if sent
+
+			const newCredential = new CredentialsEntity();
+
+			Object.assign(newCredential, req.body);
+
+			await validateEntity(newCredential);
+
+			// Add the added date for node access permissions
+			for (const nodeAccess of newCredential.nodesAccess) {
+				nodeAccess.date = new Date();
+			}
+
+			const encryptionKey = await UserSettings.getEncryptionKey();
+
+			if (!encryptionKey) {
+				throw new ResponseHelper.ResponseError(
+					RESPONSE_ERROR_MESSAGES.NO_ENCRYPTION_KEY,
+					undefined,
+					500,
+				);
```

Comment:
`UserSettings` is in /core and the encryption key error message is in /cli. I agree that this error belongs upstream, but all of the other calls to `getEncryptionKey()` in /cli repeat this pattern, e.g. in CLI commands and `WorkflowExecuteAdditionalData.ts`. Should we refactor this now? Since this is cross-package and it would affect out-of-scope files without test coverage, there might be a risk of breaking unrelated functionality.